docs/sql-data-sources-csv.md

```diff
@@ -0,0 +1,54 @@
+---
+layout: global
+title: CSV Files
+displayTitle: CSV Files
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+<div class="codetabs">
+
+<div data-lang="scala"  markdown="1">
+
+Spark SQL provides `spark.read.csv("file_name")` to read a CSV file into Spark DataFrame and `dataframe.write.csv("path")` to save or write to the CSV file. Function `option()` provides customized behavior during reading csv files. The customized behavior includes but not limit to configuring header, delimiter, charset. The detailed usage about `option()` can be found in [spark-csv](https://github.com/databricks/spark-csv), notice that some contents might be outdated in spark-csv.
+
+`spark.read.csv()` also can be used to read all files in a directory by passing a directory path. Please notice that Spark will not check the file name extension and all files will be read into dataframe.
+
+{% include_example csv_dataset scala/org/apache/spark/examples/sql/SQLDataSourceExample.scala %}
+
+</div>
+
+<div data-lang="java"  markdown="1">
+
+Spark SQL provides `spark.read().csv("file_name")` to read a CSV file into Spark DataFrame and `dataframe.write().csv("path")` to save or write to the CSV file. Function `option()` provides customized behavior during reading csv files. The customized behavior includes but not limit to configuring header, delimiter, charset. The detailed usage about `option()` can be found in [spark-csv](https://github.com/databricks/spark-csv), notice that some contents might be outdated in spark-csv.
```

Comment:
Don't duplicate this, I think? it's not language-specific. Keep it outside the language-specific blocks, once.

Reply:
Yeah, I think we don't need to mention about spark-csv here. Also, the options will be documented later via SPARK-34494

Reply:
+1 for the comment from @srowen .

@twoentartian, How about write the contents simply such as

"Spark SQL support read and write methods for CSV file, to read a CSV file into Spark DataFrame and to write the Spark DataFrame to the CSV file. `option()` can be used to customize behavior of reading or writing, such as controlling behavior of the header, delimiter character, character set, and so on."

outside of the language-specific block, after then maybe we can just simply show the code example for each language.

Refer to the "Hive Tables" page for the reference.

![Screen Shot 2021-03-30 at 6 02 29 PM](https://user-images.githubusercontent.com/44108233/112963258-28219e80-9182-11eb-9821-078aafd81959.png)

Reply:
I changed it to this :
![image](https://user-images.githubusercontent.com/13957210/113461840-93879c80-941e-11eb-97c8-f2348759171b.png)

Reply:
> Yeah, I think we don't need to mention about spark-csv here. Also, the options will be documented later via [SPARK-34494](https://issues.apache.org/jira/browse/SPARK-34494)

In addition, it seems the `option()` will be documented separately in the future, so I leave a comment in the markdown file like this:
![image](https://user-images.githubusercontent.com/13957210/113461990-2c1e1c80-941f-11eb-990e-446a9e6527e2.png)